### ISSUE TITLE
Dplyr verbs fail with parm in expression

### ISSUE THREAD

**Issue by Hong-Revo** [cloudyr/AzureKusto#42]
A raw parameterised query works:

```r
endp <- kusto_database_endpoint(server="server")
run_query(endp, "iris | where species == parm", parm="setosa")
```

The same query as a dplyr verb fails with the error "unknown column parm":

```r
ir_parm <- tbl(endp, "iris", parm="setosa")
filter(ir_parm, species == parm)
```

**Comment by Hong-Revo** [comment]
@alexkyllo is this expected behaviour? Seems to me that the dplyr expression should work just like the regular query.

**Comment by alexkyllo** [comment]
The tbl_kusto object doesn't store the parameter as a name in the data mask environment, but it does look for variables in the enclosing scope. so the dplyr example would only work if `parm` were defined as variable. I would expect this to work:

```r
parm <- "setosa"
ir_parm <- tbl(endp, "iris", parm=parm)
filter(ir_parm, species == parm)
```

**Comment by Hong-Revo** [comment]
Is it possible to change it so that the original query works? Having to define a separate variable is rather counterintuitive.

**Comment by alexkyllo** [comment]
Yes it is possible, any parameters passed would need to be stored somewhere in the `tbl_kusto` data structure and handled appropriately by the methods that lookup names in the environment--I think for example `partial_eval` would need to be enhanced.

**Comment by Hong-Revo** [comment]
OK, sounds rather complex. Do you want to handle this? Or I can give it a go if you have other stuff on your plate.

**Comment by alexkyllo** [comment]
I can do it

**Comment by Hong-Revo** [comment]
Closed in #43
